gammapy/utils/fitting/datasets.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from astropy.utils import lazyproperty
+from .parameter import Parameters
+
+
+class Datasets:
+    """Join multiple datasets
+
+    Parameters
+    ----------
+    datasets : `Dataset` or list of `Dataset`
+        List of `Dataset` objects ot be joined.
+    mask : `~numpy.ndarray`
+        Global fitting mask used for all datasets.
+
+    """
+    def __init__(self, datasets, mask=None):
+        if not isinstance(datasets, list):
+            datasets = [datasets]
+        self.datasets = datasets
+
+        if mask is not None and not self.is_all_same_type:
+            raise ValueError("Cannot apply mask if datasets are not of the same type.")
+
+        self.mask = mask
+
+        # join parameter lists
+        parameters = []
+        for dataset in self.datasets:
+            parameters += dataset.parameters.parameters
+        self.parameters = Parameters(parameters)
+
+    @property
+    def model(self):
+        return self.datasets[0].model
```

Comment:
Since the `model` is not attached to the list of `Dataset` but to each `Dataset`in particular, why would you return the first of the list?
OK I understand, this is for compatibility reasons.

Reply:
See answer to @AtreyeeS above.